File: src/main/java/spring/webmvc/domain/model/enums/DeliveryStatus.java
package spring.webmvc.domain.model.enums;

import java.util.Arrays;

import com.fasterxml.jackson.annotation.JsonCreator;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum DeliveryStatus {
	READY("준비"),
	CANCEL("취소"),
	COMP("완료");

	private final String description;

	@JsonCreator
	public static DeliveryStatus of(String name) {
		return Arrays.stream(DeliveryStatus.values())
			.filter(status -> status.name().equals(name))
			.findFirst()
			.orElse(null);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/enums/DeliveryStatus.kt
package spring.web.kotlin.domain.model.enums

import com.fasterxml.jackson.annotation.JsonCreator

enum class DeliveryStatus(
    val description: String
) {
    READY("준비"),
    CANCEL("취소"),
    COMP("완료");

    companion object {
        @JsonCreator
        @JvmStatic
        fun of(name: String) = entries.find { it.name == name }
    }
}